vanilla-todo/src/core/Component.js

```diff
@@ -0,0 +1,62 @@
+import { frozen } from "../util/fronzen.js";
+import { observable, observe } from "./observer.js";
+
+class Component {
+  $root;
+  #state;
+  #props;
+  constructor($root, props) {
+    this.$root = $root;
+    this.#props = props;
+    this.#state = observable(this.initState());
+    observe(() => {
+      this.#render();
+      this.#setEvent();
+      this.mounted();
+    });
+  }
+  get state() {
+    return frozen({ ...this.#state });
+  }
+  get props() {
+    return frozen({ ...this.#props });
+  }
+  setState(newState) {
+    for (const [key, value] of Object.entries(newState)) {
+      if (!this.#state.hasOwnProperty(key)) continue;
+      this.#state[key] = value;
+    }
+  }
+  #render() {
+    this.$root.innerHTML = this.html() || "";
+  }
+  #addEvent(eventType, targetSelector, callback) {
+    const listener = (event) => {
+      if (!event.target.closest(targetSelector)) {
+        return;
+      }
+      callback(event);
+    };
+
+    if (this.$root) {
+      this.$root.addEventListener(eventType, listener);
+    }
+  }
+  #setEvent() {
+    const events = this.event();
+    if (!events) {
+      return;
+    }
+    events.forEach((event) => {
+      this.#addEvent(event.type, event.target, event.handler);
+    });
```

Comment:
```suggestion
    this.event()?.forEach((event) => {
      this.#addEvent(event.type, event.target, event.handler);
    });
```
혹은 이렇게 표현할 수도 있겠네요!

여기서 느껴지는게, `event`라는 메소드보단 `generateEvents` 혹은 `makeEvents` `createEvents` 처럼 이벤트를 만들어준다는 느낌의 이름으로 표현하면 어떨까 싶어요!